packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/incomplete-broker-details-test.js

```diff
@@ -0,0 +1,136 @@
+   *
+   * @param {RequestState} state
+   * @param {FlowHelper} state
+   */
+  function doFetchOpportunitiesAndGetMatches(state, flow) {
+    beforeAll(async () => {
+      await state.fetchOpportunities(orderItemCriteria);
+    });
+
+    describe('Get Opportunity Feed Items', () => {
+      (new GetMatch({
+        state, flow, logger, orderItemCriteria,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+  }
+
+  /**
+   * @param {() => ChakramResponse} getChakramResponse This is wrapped in a
+   *   function because the actual response won't be available until the
+   *   asynchronous before() block has completed.
+   */
+  function itShouldReturnAnIncompleteBrokerDetailsError(getChakramResponse) {
+    it('should return an IncompleteBrokerDetailsError', () => {
+      const chakramResponse = getChakramResponse();
+      expect(chakramResponse.response).to.have.property('statusCode', 400);
+      expect(chakramResponse.body).to.have.property('@type', 'IncompleteBrokerDetailsError');
+      expect(chakramResponse.body).to.have.property('@context');
+    });
+  }
+
+  describe('Incomplete Broker Details at C1', () => {
+    const state = new RequestState(logger, { c1ReqTemplateRef: 'noBrokerName' });
+    const flow = new FlowHelper(state);
+
+    doFetchOpportunitiesAndGetMatches(state, flow);
+
+    describe('C1', () => {
+      (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup();
+
+      itShouldReturnAnIncompleteBrokerDetailsError(() => state.c1Response);
+    });
+  });
+
+  describe('Incomplete Broker Details at C2', () => {
+    const state = new RequestState(logger, { c2ReqTemplateRef: 'noBrokerName' });
+    const flow = new FlowHelper(state);
+
+    doFetchOpportunitiesAndGetMatches(state, flow);
+
+    describe('C1', () => {
+      (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('C2', () => {
+      (new C2({
+        state, flow, logger,
+      }))
+        .beforeSetup();
+
+      itShouldReturnAnIncompleteBrokerDetailsError(() => state.c2Response);
+    });
+  });
+
+  describe('Incomplete Broker Details at B', () => {
+    const state = new RequestState(logger, { bReqTemplateRef: 'noBrokerName' });
+    const flow = new FlowHelper(state);
+
+    doFetchOpportunitiesAndGetMatches(state, flow);
+
+    describe('C1', () => {
+      (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('C2', () => {
+      (new C2({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('B', () => {
+      (new B({
+        state, flow, logger,
+      }))
+        .beforeSetup();
```

Comment:
This should still run `.validationTests();` to validate the error response (in case it includes extra properties, etc)